platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -3928,8 +3932,18 @@ int odp_tm_node_shaper_config(odp_tm_node_t tm_node,
 		return -1;
 
 	odp_ticketlock_lock(&tm_glb->profile_lock);
+	if (!tm_glb->main_loop_running) {
+		tm_shaper_config_set(tm_system, shaper_profile,
+				     &tm_node_obj->shaper_obj);
+		goto out;
+	}
```

Comment:
Goto is not needed. Simple If-else should be fine.

Even simpler would be to read main_loop_running once, and then based on that call signal_req/_done.